### ISSUE TITLE
Performance Discrepancy between Double Real->Hermitian and Hermitian -> Real

### ISSUE THREAD

**Issue by dasnapkinman** [clMathLibraries/clFFT#168]
I have two plans setup, the first:

    err = clfftSetPlanPrecision(planHandle, CLFFT_DOUBLE);
    err = clfftSetLayout(planHandle, CLFFT_REAL, CLFFT_HERMITIAN_INTERLEAVED);
    err = clfftSetResultLocation(planHandle, CLFFT_INPLACE);
    err = clfftSetPlanBatchSize(planHandle, 1);

The second
    err = clfftSetPlanPrecision(planHandle2, CLFFT_DOUBLE);
    err = clfftSetLayout(planHandle2, CLFFT_HERMITIAN_INTERLEAVED, CLFFT_REAL);
    err = clfftSetResultLocation(planHandle2, CLFFT_INPLACE);
    err = clfftSetPlanBatchSize(planHandle2, 1);

   I am benchmarking as follows

    clock_gettime(CLOCK_MONOTONIC, &start);
for (int i=0;i<1000; i++) {
    err = clfftEnqueueTransform(planHandle, CLFFT_FORWARD, 1, &queue, 0, NULL, NULL, &bufX,  NULL, NULL);
    err = clfftEnqueueTransform(planHandle2, CLFFT_BACKWARD, 1, &queue, 0, NULL, NULL, &bufX, NULL, NULL);
}

    /* Wait for calculations to be finished. */
    err = clFinish(queue);
    clock_gettime(CLOCK_MONOTONIC, &end);

For benchmarking, I perform 1000 iterations of a 2048K FFT forward using the first plan and then backwards using the second. 

On my M370X this takes over 180 seconds. However if I comment out the CLFFT_BACKWARD transform the time taken is less than 10.49 seconds. Is this expected? If I use CLFFT_COMPLEX_INTERLEAVED the time goes from 14.52 seconds FORWARD only to 31.2 FORWARD->BACKWARD as expected.

**Comment by tingxingdong** [comment]
Can you rephrase this paragraph? It sounds confusing for me. 

I see you have plan-1, real-to-hermitian, plan-2, hermitian-to-real. Are you comparing plan-1 and plan-2?  Is CLFFT_COMPLEX_INTERLEAVED  a plan-3 here?


/
On my M370X this takes over 180 seconds. However if I comment out the CLFFT_BACKWARD transform the time taken is less than 10.49 seconds. Is this expected? If I use CLFFT_COMPLEX_INTERLEAVED the time goes from 14.52 seconds FORWARD only to 31.2 FORWARD->BACKWARD as expected.
/
